private/buf/cmd/protoc-gen-buf-lint/lint.go

```diff
@@ -26,48 +26,29 @@ import (
 	"github.com/bufbuild/buf/private/bufpkg/bufanalysis"
 	"github.com/bufbuild/buf/private/bufpkg/bufcheck/buflint"
 	"github.com/bufbuild/buf/private/bufpkg/bufimage"
-	"github.com/bufbuild/buf/private/pkg/app"
 	"github.com/bufbuild/buf/private/pkg/encoding"
-	"github.com/bufbuild/buf/private/pkg/protoplugin"
 	"github.com/bufbuild/buf/private/pkg/tracing"
 	"github.com/bufbuild/buf/private/pkg/zaputil"
-	"google.golang.org/protobuf/types/pluginpb"
+	"github.com/bufbuild/protoplugin"
 )
 
 const defaultTimeout = 10 * time.Second
 
 // Main is the main.
 func Main() {
-	protoplugin.Main(
-		context.Background(),
-		protoplugin.HandlerFunc(
-			func(
-				ctx context.Context,
-				container app.EnvStderrContainer,
-				responseWriter protoplugin.ResponseBuilder,
-				request *pluginpb.CodeGeneratorRequest,
-			) error {
-				return handle(
-					ctx,
-					container,
-					responseWriter,
-					request,
-				)
-			},
-		),
-	)
+	protoplugin.Main(protoplugin.HandlerFunc(handle))
 }
 
 func handle(
 	ctx context.Context,
-	container app.EnvStderrContainer,
-	responseWriter protoplugin.ResponseBuilder,
-	request *pluginpb.CodeGeneratorRequest,
+	pluginEnv protoplugin.PluginEnv,
```

Comment:
I do wonder if it weren't for this refactor, if we would have excluded passing "plugin env" in Handle, i.e.,

```diff
	Handle(
		ctx context.Context,
-		pluginEnv PluginEnv,
		responseWriter ResponseWriter,
		request Request,
	) error
```

Since most implementations would call `os.Environ()` or `os.Stderr` directly, and protoplugin is just proxying these values at this point.

Although on second thought, maybe it's a nice property the caller's function contains all the necessary bits without having to call into `os` env.